state/protocol/badger/mutator_test.go

```diff
@@ -632,40 +683,46 @@ func TestExtendHighestSeal(t *testing.T) {
 		err := state.Mutate().Bootstrap(block1, result, seal)
 		require.NoError(t, err)
 
-		// create block2 and block3
+		// create and insert block2
 		block2 := unittest.BlockWithParentFixture(block1.Header)
 		block2.Payload.Guarantees = nil
 		block2.Header.PayloadHash = block2.Payload.Hash()
 		err = state.Mutate().Extend(&block2)
 		require.Nil(t, err)
 
+		// create and insert block3 containing a receipt for block2
+		block2Receipt := unittest.ReceiptForBlockFixture(&block2)
 		block3 := unittest.BlockWithParentFixture(block2.Header)
-		block3.Payload.Guarantees = nil
-		block3.Header.PayloadHash = block3.Payload.Hash()
+		block3.SetPayload(flow.Payload{
+			Receipts: []*flow.ExecutionReceipt{block2Receipt},
+		})
 		err = state.Mutate().Extend(&block3)
 		require.Nil(t, err)
 
-		// create seals for block2 and block3
-		seal2 := unittest.SealFixture(
-			unittest.SealWithBlockID(block2.ID()),
-		)
-		seal3 := unittest.SealFixture(
-			unittest.SealWithBlockID(block3.ID()),
-		)
-
-		// include the seals in block4
+		// create and insert block4 containing a receipt for block3
+		block3Receipt := unittest.ReceiptForBlockFixture(&block3)
 		block4 := unittest.BlockWithParentFixture(block3.Header)
-		block4.Payload.Guarantees = nil
 		block4.SetPayload(flow.Payload{
+			Receipts: []*flow.ExecutionReceipt{block3Receipt},
+		})
+		err = state.Mutate().Extend(&block4)
+		require.Nil(t, err)
+
+		// create seals for block2 and block3
+		seal2 := unittest.SealFixture(unittest.SealFromResult(&block2Receipt.ExecutionResult))
+		seal3 := unittest.SealFixture(unittest.SealFromResult(&block3Receipt.ExecutionResult))
+
+		// include the seals in block5
+		block5 := unittest.BlockWithParentFixture(block4.Header)
+		block5.SetPayload(flow.Payload{
 			// placing seals in the reversed order to test
 			// Extend will pick the highest sealed block
```

Comment:
From my perspective, verifying this expected behaviour is at the heart of this test. Would you mind just adding a quick goDoc to the test? Thanks